test/rules/no-duplicate-super/test.ts.lint

```diff
@@ -0,0 +1,40 @@
+class A {
+    constructor() {
+        super();
+        super();
+        ~~~~~ [0]
+    }
+}
+
+class B {
+    constructor() {
+        super();
+        super.foo();
+    }
+}
+
+class C {
+    constructor() {
+        super();
+
+        class C2 {
+            constructor() {
+                super();
+            }
+        }
+    }
+}
+
+class D {
+    // Unfortunately, we warn here too.
+    constructor() {
+        if (cond()) {
+            super(true);
+        } else {
+            super(false);
+            ~~~~~ [0]
```

Comment:
I think this one should be doable. This could result in many false positives in real world code.

The following would be more challenging and I don't know if we want to support code like that:
```typescript
constructor() {
  if (cond)
    super(true);
  if (!cond)
    super(false);
}
```

And then there is this nasty but totally valid constructor:
```typescript
constructor() {
  if(cond)
    return super(foo);
  if(someOtherCondition)
    return super(bar);
  return super(baz);